dirty_memory_manager.cc

```diff
@@ -107,25 +108,23 @@ region_group::start_releaser(scheduling_group deferred_work_sg) {
 
 future<> region_group::release_queued_allocations() {
     {
-        return yield().then([this] {
-            return repeat([this] () noexcept {
-                if (_shutdown_requested) {
-                    return make_ready_future<stop_iteration>(stop_iteration::yes);
-                }
-
+        {
+            while (!_shutdown_requested) {
                 if (!_blocked_requests.empty() && execution_permitted()) {
                     execute_one();
-                    return make_ready_future<stop_iteration>(stop_iteration::no);
+                    co_await coroutine::maybe_yield();
                 } else {
+                  // We want `rl` to hold for the call to _relief.wait(), but not to wait
+                  // for the future to resolve, hence the inner lambda.
```

Comment:
Can also be done without a lambda:
```c++
future<> fut = make_ready_future<>();
{
    tracker_reclaimer_lock rl(logalloc::shard_tracker());
    fut = _relief.wait();
}
co_await std::move(fut);
```

Reply:
I like the lambda better since it avoids the artificial variable.